src/screens/WalletConnect/CallRequest/TransactionRequestContent.js

```diff
@@ -170,11 +170,31 @@ const useTransactionFee = (request: WalletConnectCallRequest) => {
   };
   const hasNotEnoughGas = !isEnoughBalanceForTransactionFee(walletBalances, balanceCheckTransaction, chain);
 
-  const { estimateCallRequestTransaction } = useWalletConnect();
+  const { estimateCallRequestTransaction, callRequests } = useWalletConnect();
 
-  React.useEffect(() => estimateCallRequestTransaction(request), [request, estimateCallRequestTransaction]);
+  React.useEffect(() => {
+    // perform additional check to avoid estimate on request dismiss
+    const requestExists = callRequests.some(({ callId }) => callId === request.callId);
+    if (!requestExists) return;
 
-  return { fee, gasSymbol, gasAddress, hasNotEnoughGas, isEstimating, estimationErrorMessage };
+    estimateCallRequestTransaction(request);
+  }, [request, estimateCallRequestTransaction, callRequests]);
+
+  return React.useMemo(() => ({
```

Comment:
There is no point in using `React.useMemo` here, because the only current use-case applies destructuring to the call and does not use the whole returned object.

Reply:
On one hand since this `useTransactionFee` is performing lots of operations maybe it's good to memoize it, but on the other hand I realized since it has `useEffect` in it then it won't perform estimate as return is memoized based on other values. For now removed, but something to think about it here as well.

Reply:
Ok, removing for now, but I think this needs rework.